File: MagicLibraryComponentsTest/SocketManager.m
//
//  SockerManager.m
//  RotateTogether
//
//  Created by Indieg0 on 22.06.17.
//  Copyright © 2017 None. All rights reserved.
//

#import "SocketManager.h"


@implementation SocketManager

+ (instancetype)shared {
    static SocketManager *shared = nil;
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        shared = [[SocketManager alloc] init];
    });
    return shared;
}

- (BOOL)isConnected {
    return self.socket.isConnected;
}

- (instancetype)init {
    self = [super init];
    if (self) {
        self.socket = [[JFRWebSocket alloc] initWithURL:[NSURL URLWithString:@""] protocols:nil];
        self.socket.delegate = self;
    }
    return self;
}

- (void)dealloc {
    self.socket.delegate = nil;
}

#pragma mark - Helper Methods

- (void)connect {
    [self.socket connect];
}

- (void)disconnect {
    [self.socket disconnect];
}

- (void)sendMessage:(NSString*)message {
    [self.socket writeString:message];
}

- (void)sendData:(NSData*)data {
    [self.socket writeData:data];
}

#pragma mark - JFRWebSocketDelegate

- (void)websocketDidConnect:(JFRWebSocket*)socket {
    
}

- (void)websocketDidDisconnect:(JFRWebSocket*)socket error:(NSError*)error {
    
}

- (void)websocket:(JFRWebSocket*)socket didReceiveMessage:(NSString*)string {
    
}

- (void)websocket:(JFRWebSocket*)socket didReceiveData:(NSData*)data {
    
}

@end

File: MagicLibraryComponentsTest/SocketManager.swift
//
//  SocketManager.swift
//  RotateTogether
//
//  Created by Indieg0 on 22.06.17.
//  Copyright © 2017 None. All rights reserved.
//

import UIKit
import Starscream

class SocketManager: WebSocketDelegate {

    static let shared = SocketManager()
    var socket = WebSocket(url: URL.init(string: "")!)
    
    var isConnected: Bool {
        return socket.isConnected
    }
    
    private init() {
        socket.delegate = self
    }
    
    deinit {
        socket.delegate = nil
    }
    
    //MARK: - Helper Methods
    
    func connect() {
        socket.connect()
    }
    
    func disconnect() {
        socket.disconnect()
    }
    
    func sendMessage(_ message: String) {
        socket.write(string: message)
    }
    
    func sendData(_ data: Data) {
        socket.write(data: data)
    }
    
    //MARK: - WebSocketDelegate

    func websocketDidConnect(socket: WebSocket) {
        print("websocketDidConnect")
    }
    
    func websocketDidDisconnect(socket: WebSocket, error: NSError?) {
        print("websocketDidDisconnect")
    }
    
    func websocketDidReceiveMessage(socket: WebSocket, text: String) {
        print("websocketDidReceiveMessage")
    }
    
    func websocketDidReceiveData(socket: WebSocket, data: Data) {
        print("websocketDidReceiveData")
    }
    
}
